mtprotoproxy.py

```diff
@@ -1232,7 +1226,13 @@ def gen_x25519_public_key():
         if proto_tag not in (PROTO_TAG_ABRIDGED, PROTO_TAG_INTERMEDIATE, PROTO_TAG_SECURE):
             continue
 
-        if config.SECURE_ONLY and proto_tag != PROTO_TAG_SECURE:
+        if proto_tag == PROTO_TAG_ABRIDGED and not config.OPERATING_MODES["unsecure"]:
+            continue
+
+        if proto_tag == PROTO_TAG_SECURE and not is_tls_handshake and not config.OPERATING_MODES["secure"]:
```

Comment:
For unknown reason tls mode connection also has proto_tag == PROTO_TAG_SECURE instead of PROTO_TAG_INTERMEDIATE.

So there's some kind of hack here and in line 1235 to differ secure and tls modes.